### PR TITLE
Component/map

### PR DESCRIPTION
**Pull request by julialoeschel**
- add map
- add geocoding

https://capstone-pro-component--xqtmte.herokuapp.com/

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -4,7 +4,17 @@
...
     <title>Vite App</title>
+    <script src="https://api.tiles.mapbox.com/mapbox-gl-js/v2.6.0/mapbox-gl.js"></script>
+    <link
+      href="https://api.tiles.mapbox.com/mapbox-gl-js/v2.6.0/mapbox-gl.css"
+      rel="stylesheet "
+    />
+    <!-- <link
+      href="https://api.mapbox.com/mapbox-gl-js/v2.6.0/mapbox-gl.css"
+      rel="stylesheet"
+    /> -->
```
Comment (lajanzen) [inline index.html:17]: Do you need this?

File: package.json [hunk 2]
```diff
@@ -22,6 +22,9 @@
     "@storybook/addon-links": "^6.4.0-alpha.32",
     "@storybook/react": "^6.4.0-alpha.32",
     "@types/express": "^4.17.13",
+    "@types/google.maps": "^3.47.0",
```
Comment (derjubu) [inline package.json:25]: You could remove this

**Review by derjubu — APPROVED** [review]
Looks fine to me :-) Maybe you could have made two separate PRs, on for the map, one for the geo coder, but maybe that would have been too much overhead.

File: src/app/components/Map/Map.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import 'mapbox-gl/dist/mapbox-gl.css'
+import React, { useEffect, useRef, useState } from 'react'
+import type { Map } from 'mapbox-gl'
+import mapboxgl from 'mapbox-gl' // or "const mapboxgl = require('mapbox-gl');"
```
Comment (lajanzen) [inline src/app/components/Map/Map.tsx:4]:
```suggestion
import mapboxgl from 'mapbox-gl'

**Review by lajanzen — COMMENTED** [review]
Looks like your deployment failed because there was no VITE_MAPBOX_ACCESSKEY provided. If you have the key in your .env, heroku doesn't know the variable and you have to add them to your heroku (in your review app and the prduction):

image
image

File: index.html (hunk 1 above)
Comment (marwinburesch) [inline index.html:17]: 🤔

File: package.json (hunk 2 above)
Comment (marwinburesch) [reply package.json:25]: You _should_

File: src/app/components/Map/Map.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current as HTMLElement,
+      style: 'mapbox://styles/mapbox/streets-v11',
+      center: [lng, lat],
+      zoom: zoom,
+    })
+
+    const geocoder = new MapboxGeocoder({
+      accessToken: mapboxgl.accessToken,
+      //marker: new mapboxgl.Marker({ color: 'orange' }),
```
Comment (marwinburesch) [inline src/app/components/Map/Map.tsx:33]: ಠ_ಠ

**Review by marwinburesch — COMMENTED** [review]
I like :)
[You can set environment variables in heroku](https://devcenter.heroku.com/articles/config-vars#using-the-heroku-dashboard) too to make the deployment work, I would appreciate a link to the deployment in the PR description as well :)

**Pull request merged**